packages/tools/devtools/devtools-view/src/components/data-visualization/EditableView.tsx

```diff
@@ -0,0 +1,384 @@
+	const { node, containerKey, label } = props;
+
+	/**
+	 * State to store whether the user is currently editing data
+	 */
+	const [editing, setEditing] = React.useState<boolean>(false);
+
+	/**
+	 * State to store the current type, or the type of the intended edit
+	 */
+	const [type, setType] = React.useState<string>(typeof node.value);
+
+	/**
+	 * State to store the corresponding UI component
+	 */
+	const [editingComponent, setEditingComponent] = React.useState<React.ReactElement>(<></>);
+
+	const messageRelay = useMessageRelay();
+
+	/**
+	 * On blur will set {@link editing} to false
+	 */
+	const onBlur = React.useCallback((event: React.FocusEvent<HTMLDivElement>) => {
+		/**
+		 * This allows the parent element to share focus with children
+		 * so the blur event only triggers if focus is outside of div and children
+		 */
+		if (!event.currentTarget.contains(event.relatedTarget)) {
+			setEditing(false);
+		}
+	}, []);
+
+	/**
+	 * Updates {@link editing} state to true on focus
+	 */
+	const onFocus = React.useCallback(() => {
+		setEditing(true);
+	}, []);
+
+	/**
+	 * Using the {@link fluidObjectId}, {@link containerKey}, and data posts a message to edit the DDS
+	 * @param data - The data to edit the DDS with
+	 */
+	const submitChange = React.useCallback(
+		(data: Serializable<unknown> | Primitive): void => {
+			setEditing(false);
+			const edit = {
+				fluidObjectId: node.fluidObjectId,
+				data,
+			};
+			messageRelay.postMessage(
+				DataEdit.createMessage({
+					containerKey,
+					edit,
+				}),
+			);
+		},
+		[containerKey, messageRelay, node.fluidObjectId],
+	);
+
+	/**
+	 * Determines the Editing UI based on the {@link "type"}
+	 */
+	const selectEditUI = React.useCallback(
+		(typeParamater: string): React.ReactElement => {
+			let component: React.ReactElement = <></>;
+			switch (typeParamater) {
+				case "string":
+					component = (
+						<EditableTextComponent
+							node={node}
+							setEditing={setEditing}
+							submitChange={submitChange}
+						/>
+					);
+					break;
+				case "number":
+					component = (
+						<EditableNumberComponent
+							node={node}
+							setEditing={setEditing}
+							submitChange={submitChange}
+						/>
+					);
+					break;
+				case "boolean":
+					component = (
+						<EditableBooleanComponent node={node} submitChange={submitChange} />
+					);
+					break;
+				case "undefined":
+					break;
+
+				case "null":
+					break;
+
+				default:
+			}
+
+			return component;
+		},
+		[node, submitChange],
+	);
+
+	/**
+	 * Background updates to keep the UI in sync with the data
+	 */
+	React.useEffect(() => {
+		if (editing === false) {
+			setType(typeof node.value);
+			// setLocalData(node.value);
+			setEditingComponent(selectEditUI(typeof node.value));
+		}
+	}, [node.value, editing, selectEditUI]);
+
+	/**
+	 * Updates {@link "type"} and {@link editingComponent} based on the option selected from the dropdown
```

Comment:
We should update this to reflect the behavior in the case of `null` and `undefined`.